src/test/java/com/solvd/laba/carina/mobiletests/SauceLabsTest.java

```diff
@@ -0,0 +1,86 @@
+package com.solvd.laba.carina.mobiletests;
+
+import com.solvd.laba.carina.saucelabs.common.*;
+import com.solvd.laba.carina.saucelabs.common.components.NavigationMenuBase;
+import com.zebrunner.carina.core.registrar.ownership.MethodOwner;
+import com.zebrunner.carina.utils.R;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class SauceLabsTest extends AbstractSauceLabsTest {
+
+    @Test()
+    @MethodOwner(owner = "hhan")
+    public void testLoginUser() {
+        String username = R.TESTDATA.get("username");
+        String password = R.TESTDATA.get("mobilePassword");
+
+        LoginScreenBase loginScreen = initPage(LoginScreenBase.class);
+        Assert.assertTrue(loginScreen.isOpened(), "Login screen is not opened");
+        loginScreen.typeUsername(username);
+        loginScreen.typePassword(password);
+        ProductsScreenBase productsScreen = loginScreen.clickLoginButton();
+        Assert.assertTrue(productsScreen.isOpened(), "Products screen is not opened");
+    }
+
+    @Test()
+    @MethodOwner(owner = "hhan")
+    public void testAddToCart() {
+        String productTitle = R.TESTDATA.get("productTitle");
+
+        ProductDetailsScreenBase productDetailsScreen = mobileNavigationUtils.openProductDetailsScreen(productTitle);
+        productDetailsScreen.clickAddToCartButton();
+        CartScreenBase cartScreen = productDetailsScreen.clickCartButton();
+        Assert.assertTrue(cartScreen.isOpened(), "Cart screen is not opened");
+        Assert.assertTrue(cartScreen.isProductPresent(productTitle), "Product is not added to the cart");
```

Comment:
Have you tried two or more products in cart when the productTitle is offscreen and checking isProductPresent?

Reply:
I have added swipe(). Thank you.